code/game/machinery/flatpacker.dm

```diff
@@ -0,0 +1,344 @@
+/obj/machinery/flatpacker/proc/start_build()
+	. = FALSE
+	if(!inserted_board)
+		return
+	if(!materials.has_materials(needed_mats, creation_efficiency))
+		say("Not enough materials to begin production.")
+		return
+	if(print_tier > max_part_tier)
+		say("Design too complex.")
+		return
+	materials.use_materials(needed_mats, creation_efficiency)
+	playsound(src, 'sound/items/rped.ogg', 50, TRUE)
+	busy = TRUE
+
+	addtimer(CALLBACK(src, PROC_REF(finish_build), inserted_board), flatpack_time)
+	return TRUE
+
+/// turns the supplied board into a flatpack, and sets the machine as not busy
+/obj/machinery/flatpacker/proc/finish_build(board)
+	busy = FALSE
+	new /obj/item/flatpack(drop_location(), board)
+
+/obj/machinery/flatpacker/Exited(atom/movable/gone, direction)
+	. = ..()
+	if(gone == inserted_board)
+		inserted_board = null
+		needed_mats = null
+		print_tier = 1
+		update_appearance()
+
+/obj/machinery/flatpacker/ui_act(action, list/params)
+	. = ..()
+
+	if(.)
+		return
+
+	. = TRUE
+
+	switch(action)
+		if("build")
+			if(busy)
+				return FALSE
+			start_build()
+			return TRUE
+
+		if("ejectBoard")
+			inserted_board.forceMove(drop_location())
+			return TRUE
+
+		if("eject")
+			var/datum/material/ejecting = locate(params["ref"])
+			var/amount = text2num(params["amount"])
+			if(!isnum(amount) || !istype(ejecting))
+				return TRUE
+
+			materials.retrieve_sheets(amount, ejecting, drop_location())
+			return TRUE
+
+	return FALSE
+
+/obj/machinery/flatpacker/Destroy()
+	. = ..()
+	inserted_board = null // this could be destroyed but the relevant refactor isnt in yet
+
+/obj/item/flatpack
+	name = "flatpack"
+	desc = "A box containing a compacted packed machine. Use multitool to deploy."
+	icon = 'icons/obj/devices/circuitry_n_data.dmi'
+	icon_state = "flatpack"
+	w_class = WEIGHT_CLASS_HUGE //cart time
+	throw_range = 2
+	item_flags = SLOWS_WHILE_IN_HAND | IMMUTABLE_SLOW
+	slowdown = 2.5
+	drag_slowdown = 3.5 //use the cart stupid
+	/// The board we deploy
+	var/obj/item/circuitboard/machine/board
+
+/obj/item/flatpack/Initialize(mapload, obj/item/circuitboard/machine/board)
+	. = ..()
+	if(!isnull(board))
+		src.board = board // i got board
+		board.forceMove(src)
+		var/obj/machinery/build = initial(board.build_path)
+		name += " ([initial(build.name)])"
+
+/obj/item/flatpack/Destroy()
+	. = ..()
+	qdel(board)
+
+/obj/item/flatpack/multitool_act(mob/living/user, obj/item/tool)
+	. = ..()
+	if(isnull(board))
+		return ITEM_INTERACT_BLOCKING
+	if(!isopenturf(loc))
+		user.balloon_alert(user, "cant deploy here!")
+		return ITEM_INTERACT_BLOCKING
+	balloon_alert_to_viewers("deploying!")
+	if(!do_after(user, 1 SECONDS, target = src))
+		return ITEM_INTERACT_BLOCKING
+	new /obj/effect/temp_visual/mook_dust(loc)
+	var/obj/machinery/new_machine = new board.build_path(loc)
+	loc.visible_message(span_warning("[src] deploys!"))
+	playsound(src, 'sound/machines/terminal_eject.ogg', 70, TRUE)
+	qdel(board)
```

Comment:
```suggestion
```

the board gets deleted inside `Destroy()` when you do `qdel(src)` towards the end anyway